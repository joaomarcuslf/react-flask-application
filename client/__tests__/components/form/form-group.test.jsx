import React from 'react';
import renderer from 'react-test-renderer';
import FormGroup from '../../../app/components/form/form-group';

const getChildRecurr = (tree, root) => {
  if(tree.children) {
    return getChildRecurr(
      tree.children[root] ||
      tree.children[0],
      root
    );
  }

  return tree;
}

test('FormGroup wraps the visualization for the input showing it\'s label', () => {
  const component = renderer.create(
    <FormGroup label="Test Label" name="test" error={{}}>
      Test
    </FormGroup>,
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
  expect(getChildRecurr(tree)).toEqual('Test Label');
  expect(getChildRecurr(tree, 2)).toEqual('Test');
});
