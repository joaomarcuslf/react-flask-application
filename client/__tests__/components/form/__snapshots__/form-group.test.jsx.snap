// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup wraps the visualization for the input showing it's label 1`] = `
<label
  className="form__FormLabel-vl64uu-1 hGProG"
  for="test"
>
  Test Label
  :
  Test
</label>
`;
